rma/models/rma.py

```diff
@@ -0,0 +1,715 @@
+                        move.product_uom_qty, record.reception_uom)
+            record.received_qty = qty
+
+    @api.depends(
+        'refund_id.state',
+        'refund_line_id.quantity',
+    )
+    def _compute_refunded_qty(self):
+        for record in self:
+            refund_line = record.refund_line_id
+            if record.refund_id.state not in ['draft', 'cancel']:
+                record.refunded_qty = refund_line.uom_id._compute_quantity(
+                    refund_line.quantity, record.reception_uom)
+            else:
+                record.refunded_qty = 0.0
+
+    @api.multi
+    @api.depends(
+        'delivery_move_ids.state',
+        'delivery_move_ids.scrapped',
+        'delivery_move_ids.product_uom_qty',
+        'delivery_move_ids.product_uom',
+    )
+    def _compute_delivered_qty(self):
+        """
+        Based on:
+        https://github.com/OCA/OCB/blob/12.0/addons/sale_stock/models/
+        sale_order.py#L203
+        :return:
+        """
+        for record in self:
+            qty = 0.0
+            product = record.delivery_product_id
+            for move in record.delivery_move_ids.filtered(
+                    lambda r: (r.state == 'done' and not r.scrapped
+                               and r.product_id == product)):
+                usage = move.location_dest_id.usage
+                if usage == "customer":
+                    if (not move.origin_returned_move_id
+                            or (move.origin_returned_move_id
+                                and move.to_refund)):
+                        qty += move.product_uom._compute_quantity(
+                            move.product_uom_qty, record.delivery_uom)
+                elif usage != "customer" and move.to_refund:
+                    qty -= move.product_uom._compute_quantity(
+                        move.product_uom_qty, record.delivery_uom)
+            record.delivered_qty = qty
+
+    # Constrains (@api.constrains)
+    # onchange methods (@api.onchange)
+    @api.onchange("user_id")
+    def _onchange_user_id(self):
+        if self.user_id:
+            self.team_id = self.env['rma.team'].sudo().search([
+                '|',
+                ('user_id', '=', self.user_id.id),
+                ('member_ids', '=', self.user_id.id),
+                '|',
+                ('company_id', '=', False),
+                ('company_id', 'child_of', self.company_id.ids)
+            ], limit=1)
+        else:
+            self.team_id = False
+
+    @api.onchange("move_id")
+    def _onchange_move_id(self):
+        if self.move_id:
+            self.product_id = self.move_id.product_id
+        else:
+            self.product_id = False
+
+    @api.onchange("picking_id")
+    def _onchange_picking_id(self):
+        self.move_id = False
+        self.product_id = False
+
+    @api.onchange("product_id")
+    def _onchange_product_id(self):
+        product = self.product_id
+        domain = {
+            'reception_uom': [
```

Comment:
Maybe a good usability improvement could be to change the `picking_id` and `move_id` domains as well so only those containing the product are available.

And maybe it could be a little bit trickier to combine, but the same combining with a partner's onchange would be amazing.